src/encode/pattern/mod.rs

```diff
@@ -667,10 +727,7 @@ impl Deserialize for PatternEncoderDeserializer {
         config: PatternEncoderConfig,
         _: &Deserializers,
     ) -> Result<Box<dyn Encode>, Box<dyn Error + Sync + Send>> {
-        let encoder = match config.pattern {
-            Some(pattern) => PatternEncoder::new(&pattern),
-            None => PatternEncoder::default(),
-        };
+        let encoder: PatternEncoder = config.into();
```

Comment:
I think I prefer keeping the previous strategy of pattern is None if unspecified and right here swap in the default to log with.

Reply:
Looking back at the change, i did this b/c i also made the change in the `PatternEncoderConfig` 
```  pub struct PatternEncoderConfig {
      pattern: Option<String>,
  }```